agent/client/client.go

```diff
@@ -350,6 +362,9 @@ loop:
 			if !more {
 				break loop
 			}
+			if req == nil {
+				continue
+			}
```

Comment:
It will protect only if channel is done, but in case of nil value and opened channel it will fail on nil pointer there: https://github.com/percona/pmm/pull/2410/files#diff-4aa388772db1a19eca352e250851b7785c15d428909601614ef5cc4afb325f22R370

Should I remove it anyway?

Reply:
when we might have nil value for non-closed channel?

Reply:
We could send it there, but seems there is no place like that. Removed.